File: FactoryTs/articulofactory.ts
import { TipoArticulo } from "./tipoarticulo";
import { ArticuloInformatico } from "./articuloinformatico";
import { ArticuloOficina } from "./articulooficina";

export class ArticuloFactory{

    constructor(){

    }
    public create(idArticulo:number, 
                  nombre:string, 
                  tipoArticulo:TipoArticulo, 
                  precio:number, 
                  isExento:boolean) {
        
        if(tipoArticulo.getTipoArticulo() == TipoArticulo.TIPO_INFORMATICO){
            
            return new ArticuloInformatico(idArticulo, nombre, tipoArticulo, precio, isExento);
        }
        if(tipoArticulo.getTipoArticulo() == TipoArticulo.TIPO_OFICINA){
            return new ArticuloOficina(idArticulo, nombre, tipoArticulo, precio);
        }
        return null;
        
    }

}

File: FactoryExample/src/factoryexample/ArticuloFactory.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package factoryexample;

/**
 *
 * @author fabm_
 */
public class ArticuloFactory implements IArticuloFactoryMethod{

    @Override
    public Articulo create(int id, String nombre, TipoArticulo tipoArticulo, long precio, boolean isExento) {
        
        if(tipoArticulo.getTipoArticulo() == TipoArticulo.TIPO_INFORMATICO){
            return new ArticuloInformatico(id, nombre, tipoArticulo, precio, isExento);
        }
        if(tipoArticulo.getTipoArticulo() == TipoArticulo.TIPO_OFICINA){
            return new ArticuloOficina(id, nombre, tipoArticulo, precio);
        }
        return null;
        
    }
    
}
